### PR TITLE
Move field into place when adding during schema evolution

### PR DESCRIPTION
**Pull request by hiloboy0119**
While trying to use schema evolution on write I noticed new fields were always being added to the end of the struct/table.  This caused schema evolution to fail.  I added tests to verify and fixed by moving the field if necessary

### CODE REVIEW

**Comment by amogh-jahagirdar** [comment]
@hiloboy0119 thanks for the PR, do you mind clarifying what was the exact way you were trying to evolve the table schema via Spark ACCEPT_ANY_SCHEMA, and the specific failure? That would help a lot in terms of evaluating the PR

**Review by amogh-jahagirdar — COMMENTED** [review]
@hiloboy0119 thanks for the PR, do you mind clarifying what was the exact way you were trying to evolve the table schema via Spark ACCEPT_ANY_SCHEMA, and the specific failure? That would help a lot in terms of reviewing the PR

**Comment by hiloboy0119** [comment]
@amogh-jahagirdar sorry for the long delay.  Finally had time to write tests for the various edge cases and fix a few things.

The examples of schema evolution are all shown in test cases.  I added fields to the beginning and middle of tables.  Without these changes they were always added to the end of the schema.

File: core/src/main/java/org/apache/iceberg/schema/UnionByNameVisitor.java [hunk 1]
```diff
@@ -90,6 +94,15 @@ public Boolean struct(
               Types.NestedField field = fields.get(pos);
               if (isMissing) {
                 addColumn(partnerId, field);
```
Comment (hiloboy0119) [inline core/src/main/java/org/apache/iceberg/schema/UnionByNameVisitor.java:96]: @amogh-jahagirdar This is the core problem with the existing code.  Additional columns, no matter where they are in the schema are just added.  This always adds them to the end

**Pull request closed without merging**